src/pages/index.js

```diff
@@ -60,28 +107,122 @@ document.addEventListener("DOMContentLoaded", function () {
     addFormValidator.resetValidation();
   });
 
-  newCardList.renderItems();
+  constants.avatar.addEventListener("click", () => {
+    avatarPopup.open();
+    addFormValidator.resetValidation();
+  });
+
+  constants.avatar.addEventListener("mousemove", () => {
+    constants.avatarButton.classList.replace(
+      "profile__avatar-button_disabled",
+      "profile__avatar-button"
+    );
+  });
+});
+
+constants.avatar.addEventListener("mouseout", () => {
+  constants.avatarButton.classList.replace(
+    "profile__avatar-button",
+    "profile__avatar-button_disabled"
+  );
 });
 
 /* -------------------------------------------------------------------------- */
 /*                                  Functions                                 */
 /* -------------------------------------------------------------------------- */
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    (id) => handleDeleteClick(id),
+    (id) => handleLikeClick(id)
+  );
   return card.getView();
 }
 
-function handleImageClick(name, link) {
-  newPopupImage.open(name, link);
+function handleImageClick(cardData) {
+  newPopupImage.open(cardData);
+}
+
+function handleDeleteClick(card) {
+  popupAffirm.open();
+  popupAffirm.setSubmitAction(() => {
+    api
+      .deleteCards(card._id)
+      .then(() => {
+        popupAffirm.close();
+        card.handleDeleteCard();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleLikeClick(card) {
+  if (card.isLiked === true) {
+    api
+      .deleteCardLike(card._id)
+      .then((res) => {
+        card.isLiked = false;
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .putCardLike(card._id)
+      .then((res) => {
+        card.isLiked = true;
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
 }
 
 function handleProfileFormSubmit(formData) {
-  useInfo.setUserInfo(formData.title, formData.description);
-  editPopup.close();
+  constants.editProfileSubmitButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```